pkg/apis/flagger/v1beta1/canary.go

```diff
@@ -304,6 +304,8 @@ type CanaryMetric struct {
 	// TemplateRef references a metric template object
 	// +optional
 	TemplateRef *CrossNamespaceObjectReference `json:"templateRef,omitempty"`
+
+	Variables map[string]string `json:"variables,omitempty"`
```

Comment:
Maybe `TemplateVariables` as these only work for MetricTemplates?

Reply:
Good point. This has been updated.

Reply:
You also need to update the crd in the manifests dir, then run `make crd`.

Reply:
Thanks for that. CRD's have been updated.